qiskit/opflow/list_ops/list_op.py

```diff
@@ -72,12 +72,19 @@ def __init__(self,
             identity - it accepts the list of values, and returns them in a list.
         """
         super().__init__()
-        self._oplist = oplist
+        self._oplist = self._get_op_list(oplist)
         self._combo_fn = combo_fn
         self._coeff = coeff
         self._abelian = abelian
         self._grad_combo_fn = grad_combo_fn
 
+    def _get_op_list(self, oplist):
```

Comment:
This just seems to be type checking across the interface to fail early. Though the types its checking for are different than in the typehint of the parameter. Should that then state `oplist: List[Union[OperatorBase, float]]`, to mark this as a permissable usage or is this more a "hack" as a temp fix for the issue. If we state float in the typehint then int types would be implied to be supported but the type check would fail in that regard so maybe can allow int too? While this seems to be type check it also casts the param to a list, which is to what the stated type is in the param, hence it broadens the types allowed by doing this. Is that intentional too?

In regards of the error, since its a type check should it not be a TypeError raised with the type of the entry, maybe its value too if thats helpful.

Reply:
* Yes, it's just type checking so the interface fails early.
* Yes, the type hint should be updated.
* Yes `TypeError` is the correct error
* "If we state float in the typehint then int types would be implied ..." That's unfortunate. Well, we should make an effort to express the same meaning with the two type systems. Using a multiple (or single) dispatch library would be an advantage here because a single type specification serves both for documentation and type checking.
* I don't intend allowing `float` here to be a hack. It should be exposed.
*  "...hence it broadens the types allowed by doing this". Yes, this is intentional. I am following what the code did previously. I assume this is meant to support tuples, generators, and so forth.

Reply:
Looks like we might need something more general, so I put  this on hold.

Reply:
In regards of the typehint, and the way float should be viewed, this is part of the PEP, https://www.python.org/dev/peps/pep-0484/#the-numeric-tower I imagine a dispatch library would do the right thing typewise as part of its dispatch